src/modules/dashboard/components/CreateEditDomainDialog/CreateEditDomainDialogForm.tsx

```diff
@@ -0,0 +1,162 @@
+import { canArchitect } from '../../../users/checks';
+
+import { FormValues } from './CreateEditDomainDialog';
+import styles from './CreateEditDomainDialogForm.css';
+
+const MSG = defineMessages({
+  titleCreate: {
+    id:
+      'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.titleCreate',
+    defaultMessage: 'Create a new domain',
+  },
+  titleEdit: {
+    id: 'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.titleEdit',
+    defaultMessage: 'Edit domain details',
+  },
+  name: {
+    id: 'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.name',
+    defaultMessage: 'Domain name',
+  },
+  purpose: {
+    id: 'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.name',
+    defaultMessage: 'What is the purpose of this domain?',
+  },
+  annotation: {
+    id:
+      'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.annotation',
+    defaultMessage: 'Explain why you’re creating this domain',
+  },
+  noPermission: {
+    id:
+      // eslint-disable-next-line max-len
+      'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.noPermission',
+    defaultMessage:
+      // eslint-disable-next-line max-len
+      'You need the {roleRequired} permission in {domain} to take this action.',
+  },
+});
+
+interface Props {
+  back: () => void;
+  colony: Colony;
+  id?: string;
+}
+
+const CreateEditDomainDialogForm = ({
+  back,
+  colony,
+  handleSubmit,
+  id,
+}: Props & FormikProps<FormValues>) => {
+  const [domainColor, setDomainColor] = useState(Color.LightPink);
+
+  const { walletAddress, username, ethereal } = useLoggedInUser();
+
+  const allUserRoles = useTransformer(getAllUserRoles, [colony, walletAddress]);
+
+  const hasRegisteredProfile = !!username && !ethereal;
+  const canCreateEditDomain =
+    hasRegisteredProfile && canArchitect(allUserRoles);
+
+  return (
+    <>
+      <DialogSection>
+        <Heading
+          appearance={{ size: 'medium', margin: 'none' }}
+          text={id === undefined ? MSG.titleCreate : MSG.titleEdit}
+          className={styles.title}
+        />
+      </DialogSection>
+      {!canCreateEditDomain && (
+        <DialogSection>
+          <PermissionRequiredInfo requiredRoles={[ColonyRole.Architecture]} />
+        </DialogSection>
+      )}
+      <DialogSection>
+        <div className={styles.nameAndColorContainer}>
+          <div className={styles.domainName}>
+            <Input
+              label={MSG.name}
+              name="name"
+              appearance={{ colorSchema: 'grey', theme: 'fat' }}
+              disabled={!canCreateEditDomain}
+            />
+          </div>
+          <ColorSelect
+            activeOption={domainColor}
+            appearance={{ alignOptions: 'right' }}
+            onColorChange={setDomainColor}
+            disabled={!canCreateEditDomain}
+          />
+        </div>
+      </DialogSection>
+      <DialogSection>
+        <Input
+          label={MSG.purpose}
```

Comment:
Shouldn't this Input have a maxLength applied? From what I saw in the design, it should have a max length of 90

Reply:
Nope. See this: https://github.com/JoinColony/colonyDapp/pull/2339#discussion_r551922330

Reply:
Oh I see! Nevermind then, all good :ok_hand: